### PR TITLE
Replace com.google.common.base.Optional with Java 11 implementation

### PR DESCRIPTION
**Pull request by pks-1981**
Replace com.google.common.base.Optional

For https://github.com/wiremock/wiremock/issues/2111

### CODE REVIEW

**Comment by pks-1981** [comment]
@timtebeek @tomakehurst waiting for review)

**Review by oleg-nenashev — APPROVED** [review]
+1 from me. It is breaking binary compatibility, which is something we plan for WireMock 3.0 in the case of Guava&Co

**Comment by pks-1981** [comment]
@tomakehurst
Update after merge?
Classes RequestWrapper & WireMockApp are little corrupted...

**Pull request merged**
